.github/workflows/doc-build.yml

```diff
@@ -85,7 +85,6 @@ jobs:
     if: github.repository == 'pytorch/executorch' && github.event_name == 'push' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v'))
     permissions:
       contents: write
```

Comment:
based on [this](https://github.com/pytorch/executorch/pull/7819/files#diff-1048a3f83dfee0a99ce3c07fede976bbd6a6b65f8bdae55ada11e6a1dda452c7), should this be changed to `id-token: write` @jzhang337